plugins/action/generate_spec.py

```diff
@@ -0,0 +1,92 @@
+# (c) 2020, Ansible by Red Hat, inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from ansible.plugins.action import ActionBase
+import os
+from ansible.errors import AnsibleError
+
+from ansible.module_utils._text import to_text
+YANG_SPEC_DIR_PATH = "~/.ansible/tmp/yang_spec"
+from ansible_collections.community.yang.plugins.lookup.spec import (
+    LookupModule,
+)
+
+class ActionModule(ActionBase):
+    def run(self, terms=None, variables=None, **kwargs):
+
+        result = super(ActionModule, self).run(terms, variables)
+
+        # Read and validate yang file path
+        try:
```

Comment:
The argspec validation can be done using the validation document to argspec conversion in netcommon.
Refer https://github.com/ansible-collections/ansible.netcommon/blob/main/plugins/action/cli_parse.py#L89-L118